weaviate/connect/v4.py

```diff
@@ -411,6 +407,7 @@ def __send(
                 url,
                 json=weaviate_object,
                 params=params,
+                headers=self.__get_headers_for_async(),
```

Comment:
Can we rename this to something like `__get_latest_headers` to show why it's requested here? Using headers go into the client creation in `httpx` as above so this is a different use